### ISSUE TITLE
Why setting pad logit to -inf?

### ISSUE THREAD

**Issue by alireza202** [OpenNMT/OpenNMT-py#714]
Can you explain the logic behind this line:

`logits[:, self.tgt_dict.stoi[onmt.io.PAD_WORD]] = -float('inf')`

**Comment by sebastianGehrmann** [comment]
We never want to predict the padding token. Therefore, we set all the probabilities of the symbol to 0, or -in logits- to `log(0) = -inf`

**Comment by sebastianGehrmann** [comment]
The attention towards pad tokens is also set to 0.

**Comment by da03** [comment]
I think this issue is resolved? Please reopen it if I was wrong.
